test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -719,7 +721,12 @@ bool UcxConnection::connect_common(ucp_ep_params_t& ep_params)
     if (_use_am) {
         // With AM use ep as a connection ID. AM receive callback provides
         // reply ep, which can be used for finding a proper connection.
-        _conn_id = reinterpret_cast<uint64_t>(_ep);
+        // Since connection ids are not explicitly exchanged, set
+        // _remote_conn_id to dummy value, to handle connection errors
+        // properly in handle_connection_error().
+        _conn_id        = reinterpret_cast<uint64_t>(_ep);
+        _remote_conn_id = 0xffffffffffffffff;
```

Comment:
maybe introduce new member variable: bool _connected

Reply:
you mean instead of `_remote_conn_id`?

Reply:
it will be the indication if connection is completed, instead of _remote_conn_id!=0
(remote_conn_id will still be used for tags)